### PR TITLE
Implement CTA in Idea Hub widget for when the module is not activated (3275).

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #3275

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/sass/config/_variables.scss
Comment (aaemnnosttv) [inline assets/sass/config/_variables.scss:69]: We should probably keep color variables as normal hex codes without alpha values. We can always apply it with a different alpha, but the color itself should be just that.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss:59]: Is there a reason we need to add the icon with CSS rather than in React as we normally would?

If we _really_ needed to, we could use `{ renderToString } from '@wordpress/element'` to get a string from a React element rather than duplicating it here.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss:46]: This seems a bit large to be a fixed width, are we should that's what it should be?

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:64]: Why the leading underscore on all the classes?

**Review by aaemnnosttv — COMMENTED** [review]
Looks like a good start @johnPhillips – I know this isn't ready for a full review yet but I've left some feedback as requested 👍

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss
Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss:46]: It was just because I was working out of storybook until #3274 got merged. I've removed it now.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss
Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss:59]: I was unsure of the convention here. Happy to move it over to React 👍

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (asvinb) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:65]: @johnPhillips Can we rename the class name to `googlesitekit-idea-hub__dashboard--cta` ? This change applies to the other class names as well.

**Review by asvinb — CHANGES_REQUESTED** [review]

File: assets/sass/config/_variables.scss
Comment (asvinb) [inline assets/sass/config/_variables.scss:69]: @johnPhillips Let's move the variable a bit further down in the file where we have other `$c-idea-hub-*` variables. :)

**Review by asvinb — CHANGES_REQUESTED** [review]

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:65]: @asvinb To be fair we don't have very clear precedent to follow regarding class names I think using `--` as suggested here would be another use that I don't think we have yet. Let's keep `--` to signify state/variant. In this case I think `googlesitekit-idea-hub__dashboard-cta` is probably the least inconsistent for a base 😆

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:71]: Nit: Please define text nodes on their own line when wrapped.
```suggestion
				<h5>
					{ __( 'Get new topics based on what people are searching for with Idea Hub', 'google-site-kit' ) }
				</h5>
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:85]: Nit: Please keep braces in the same column in JSX
```suggestion
				<Button onClick={ onClick }>
					{
						active && ! connected
							? __( 'Complete set up', 'google-site-kit' )
							: __( 'Set up', 'google-site-kit' )
					}
				</Button>
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js:22]: This block is unused 🙂 
```suggestion
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js:43]: This only works because it doesn't rely on any thing registered from its datastore/module. When the test registry is created above, the `idea-hub` feature won't be enabled and so this may lead to some odd behavior/results.

This should be fixed in #2940 which is just pending review, but would require slightly different implementation here.

File: assets/sass/config/_variables.scss
Comment (aaemnnosttv) [reply assets/sass/config/_variables.scss:69]: @asvinb maybe those colors below should move up here instead since PSI colors are above? Or maybe all the module colors should move to the bottom? 😄

**Review by aaemnnosttv — COMMENTED** [review]
LGTM, just a few comments around @asvinb 's feedback and a few details for improvements.

It would be great if the stories here could build on #2940 but I'm not sure if that will be merged in time. I'll see if we can move that one a bit quicker.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss
Comment (aaemnnosttv) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub.scss]: Yeah, we should use svg components where possible as it allows much more flexibility, particularly around styling.

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (asvinb) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:65]: @aaemnnosttv I agree that we don't really have an agreed convention for class names but if we strictly follow the BEM methodologies, the class names can get very long and ugly. Maybe we can use a mix of our own methodology and BEM.  I'am thinking what's the best way to avoid having such class names in the code:

- googlesitekit-pagespeed-insights-web-vitals-metrics--field-data-unavailable
- googlesitekit-pagespeed-report-metric-value__display-value
- .googlesitekit-widget--analyticsAllTraffic__user-count-chart

File: assets/sass/config/_variables.scss
Comment (asvinb) [reply assets/sass/config/_variables.scss:69]: @aaemnnosttv Actually it's good you caught this. Ideally we should define the colors first (without any specific module prefix to it) using https://chir.ag/projects/name-that-color/
In this case it'll be : 
```
$c-pizazz: #f78f01;
```

Then we use the `$c-pizazz` variable for `$c-idea-hub-orange`

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js:43]: @aaemnnosttv Could you elaborate on this please? 

> When the test registry is created above, the idea-hub feature won't be enabled and so this may lead to some odd behavior/results.

This CTA is rendered when the feature is not enabled... maybe I am misunderstanding something. Should I be using `enabledFeatures.clear();` to ensure that it is not enabled, for example?

> This should be fixed in #2940 which is just pending review, but would require slightly different implementation here.

I'm looking through the changes in [Asvin's PR](https://github.com/google/site-kit-wp/pull/3382/files/8a453ed75bc2ddd7066eaa535a8d77e859c3e34f# ) which got merged and seems to be fairly similar. 
Am I ok to use that as a guide for this? Is there anything specific I need to include?

File: assets/sass/config/_variables.scss
Comment (johnPhillips) [reply assets/sass/config/_variables.scss:74]: Trying to summarize. @aaemnnosttv and @asvinb Are you good with this approach?
```diff
... 

 $c-shark: #202124;
 $c-oslo-gray: #80868b;
+ $c-pizazz: #f78f01;  // 1. Define new color with the others at the top

 // PageSpeed Insights colors
 $c-pagespeed-fast: $c-green-700;
 $c-pagespeed-average: $c-yellow-900;
 $c-pagespeed-slow: $c-red-600;

- // Idea Hub colors
- $c-idea-hub-orange: #f78f01;

 $c-base: $c-white;
 $c-primary: $c-boulder;

...

 $c-user-input-placeholder: $c-heather;
 $c-user-input-color: $c-denim;

+ // Idea Hub colors
+ $c-idea-hub-orange: $c-pizzaz;  // 2. Move the $c-idea-hub-orange declaration to the bottom with the other Idea Hub vars
 $c-idea-hub-background: $c-white;
 $c-idea-hub-background-hover: $c-gallery-2;
 $c-idea-hub-button-hover: $c-iron;
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:65]: @asvinb and @aaemnnosttv What's the consensus, are we happy with `googlesitekit-idea-hub__dashboard-cta`?

Happy to go with whatever you think.

File: assets/sass/config/_variables.scss
Comment (asvinb) [reply assets/sass/config/_variables.scss:74]: Works for me @johnPhillips 😁

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (asvinb) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:65]: I'll defer toe @aaemnnosttv for final decision 😁

Comment (aaemnnosttv) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:65]: @asvinb @johnPhillips Let's not get too hung up on this since we don't have a clear rule for how we write our class names so whichever way we go, we'll still be inconsistent overall. IMO a utility-first approach avoids most of these problems to begin with which is my preferred "way to avoid having such class names in the code" 😄 We will need to revisit our strategy for CSS in the not too distant future so it's not worth spending too much time crafting the right name for.

Let's go with `googlesitekit-idea-hub__dashboard-cta` for now and we can work on defining the approach going forward (even if temporary) to reduce confusion in the future.

File: assets/sass/config/_variables.scss
Comment (aaemnnosttv) [reply assets/sass/config/_variables.scss]: LGTM @johnPhillips

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js
Comment (aaemnnosttv) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.stories.js]: @johnPhillips #2940 is merged now so the key takeaways there are:

- Stories should no longer be wrapped with `WithTestRegistry` _in the story itself_ as this is applied in a global decorator now to every story
    - Features should be set on `Story.parameters.features` now instead of passing as a prop to the test registry
    - All features are be disabled by default for each story automatically, so nothing extra is needed for a story that needs a disabled feature
- Similarly, `createTestRegistry` shouldn't be used at the story level anymore – maybe there's some edge case where it's needed, but as a general rule we only need to provide registry setup for the story (see `WithRegistrySetup`)


The PR you referenced there isn't quite up to date with all of the above. Best to reference `DashboardSearchVisitorsWidget.stories.js` first and then let me know if you still have questions

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:50]: Why not set the `height` and `width` on the `BulbIcon` instead? It currently has `10` set for both. Dimensions like this are best set on the component.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:46]: I think this rule is no longer relevant since there is more background image, correct?

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:49]: Is there a variable we can use for this value? If not, why 32px specifically?

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:56]: Similar to the last comment, I think the `height` can be set on the component itself.

As for the margin between, it looks like a `&nbsp;` pretty well 
image

As for the negative margin-bottom, this seems to be just for vertical alignment. Let's use `vertical-align` or flex-based alignment rather than manually tweaking with margins instead.

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:41]: This seems a bit out of place in the component here. Ideally this should go in module-level `constants.js` but even Analytics doesn't have one for the `MODULE_SLUG` yet. Let's inline this for now but it would be great if you could open a new issue for refactoring module slug references with constants.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @johnPhillips – this is looking pretty good. I left a few more comments but nothing major. Let me know if you have any other questions 👍

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:49]: I've done something that made sense in the moment but I'm definitely more than happy to change it. I set the height and width of the SVG to 144px (I have now done this on the `BulbIcon` as you suggested), and gave it a circular `border-radius` and `background-colour` to achieve the 144px x 144px circle. 
The 32px padding then brings the icon down to the correct height:

| 144px x 144px circle | 32px padding |
| --- | --- |
| Screenshot 2021-05-26 at 10 31 08 | Screenshot 2021-05-26 at 10 29 47 


I can definitely understand that this might be hard to read. I was just trying to avoid a meaningless `<div>` or similar to get the circle.

Let me know what you think.

Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:49]: Just noticed it was actually 30px in the design, so I'll update it and add a comment explaining the use of padding. Again, happy to change the approach here.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:50]: I think my thought process was that since it is sitting inline with the text I wanted to use REM so that it increases / decreases nicely along with the font size. 
I've removed this and added the fixed height to the `BulbIcon` as suggested.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:56]: Great point about the `&nbsp;`, and I had totally forgotten about the existence of `vertical-align`... 🤦‍♂️ 
Thanks!

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js]: @aaemnnosttv I've created an issue for this - let me know what you think?
https://github.com/google/site-kit-wp/issues/3417

**Comment by aaemnnosttv** [comment]
@johnPhillips would you please resolve the merge conflict?

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:76]:
```suggestion
						width="16"
						height="16"
```

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (aaemnnosttv) [inline assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:56]: This is scrutinizing pixels here a bit but I think it looks better aligning with flexbox; there doesn't seem to be a `vertical-align` option that has the same result 😕 

<table>
<tr>
	<td><code>vertical-align</code></td>
	<td><code>flex</code></td>
<tr>
	<td>

image
image

</td>
<td>

image
image

</td>
</table>

To achieve this I just removed this one rule, and added these two to the parent `p` tag

```css
    display: flex;
    align-items: center;
```

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:53]:
```suggestion
				id: 'idea-hub-setup-error',
```

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @johnPhillips – this is really close, just a few last things to clean up, and a merge conflict to resolve and then this should be good to go 👍

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (aaemnnosttv) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss]: The style is specific to this component so it should be fine for now. If we need to use it in other places in different sizes we might want to use a percentage for the padding instead to preserve the proportions.

File: assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss
Comment (johnPhillips) [reply assets/sass/components/idea-hub/_googlesitekit-idea-hub-dashboard-cta.scss:56]: Nothing wrong with scrutinizing pixels! I'll give it a go 👍

**Review by aaemnnosttv — APPROVED** [review]
LGTM; thanks @johnPhillips !

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:1]: One thing I missed here 😬  the ACs call for a `assets/js/modules/idea-hub/components/dashboard/DashboardCTA.js` component but this uses a directory. Currently we only use this for components which are composed of sub-components which are local to the larger one. Let's update this to use the path as described in the ACs for consistency.

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (aaemnnosttv) [inline assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js:80]: This wasn't explicitly mentioned in the ACs, but as an external link this should have the `external` prop to open it in a new tab. We can keep the external icon hidden to stay consistent with the design (it also doesn't look great with an icon on both sides of the text 😄 )

```suggestion
						href="https://sitekit.withgoogle.com/documentation/idea-hub-module/"
						external
						hideExternalIndicator
					>
```

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
A few tiny things I missed before – should be super quick to address, then we'll get this merged 😄

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js]: Oops, yes. Good catch.

File: assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardCTA/index.js]: Added 👍

**Review by aaemnnosttv — APPROVED** [review]
Thanks @johnPhillips 👍

**Pull request merged**
